timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java

```diff
@@ -66,30 +85,43 @@ public static TimeLockAgent create(
             Supplier<TimeLockRuntimeConfiguration> runtime,
             TimeLockDeprecatedConfiguration deprecated,
             Consumer<Object> registrar) {
-        TimeLockAgent agent = new TimeLockAgent(metricsManager, install, runtime, deprecated, registrar);
+        ExecutorService executor = createSharedExecutor(metricsManager);
+        TimeLockAgent agent = new TimeLockAgent(metricsManager, install, runtime, deprecated, registrar, executor);
         agent.createAndRegisterResources();
         return agent;
     }
 
     private TimeLockAgent(MetricsManager metricsManager, TimeLockInstallConfiguration install,
             Supplier<TimeLockRuntimeConfiguration> runtime,
             TimeLockDeprecatedConfiguration deprecated,
-            Consumer<Object> registrar) {
+            Consumer<Object> registrar,
+            ExecutorService sharedExecutor) {
         this.metricsManager = metricsManager;
         this.install = install;
         this.runtime = runtime;
         this.registrar = registrar;
-
+        this.sharedExecutor = sharedExecutor;
         this.paxosResource = PaxosResource.create(metricsManager.getRegistry(),
                 install.paxos().dataDirectory().toString());
-        this.leadershipCreator = new PaxosLeadershipCreator(this.metricsManager, install, runtime, registrar);
+        this.leadershipCreator = new PaxosLeadershipCreator(metricsManager, install, runtime, registrar);
         this.lockCreator = new LockCreator(runtime, deprecated);
         this.timestampCreator = getTimestampCreator(metricsManager.getRegistry());
         LockLog lockLog = new LockLog(metricsManager.getRegistry(),
                 Suppliers.compose(TimeLockRuntimeConfiguration::slowLockLogTriggerMillis, runtime::get));
         this.timelockCreator = new AsyncTimeLockServicesCreator(metricsManager, lockLog, leadershipCreator);
     }
 
+    private static ExecutorService createSharedExecutor(MetricsManager metricsManager) {
+        return new InstrumentedExecutorService(
+                PTExecutors.newCachedThreadPool(
+                        new InstrumentedThreadFactory(new ThreadFactoryBuilder()
+                                .setNameFormat("paxos-timestamp-creator-%d")
+                                .setDaemon(true)
+                                .build(), metricsManager.getRegistry())),
+                metricsManager.getRegistry(),
+                MetricRegistry.name(PaxosLeaderElectionService.class, "paxos-timestamp-creator", "executor"));
+    }
```

Comment:
This is fine, I guess because there should only be one executor for the course of the JVM, and the threads spun up are daemon threads.